src/sql/platform/telemetry/telemetry.contribution.ts

```diff
@@ -18,32 +18,89 @@ export class SqlTelemetryContribution extends Disposable implements IWorkbenchCo
 	) {
 		super();
 
-		const dailyLastUseDate = Date.parse(storageService.get('telemetry.dailyLastUseDate', StorageScope.GLOBAL, '0'));
-		const weeklyLastUseDate = Date.parse(storageService.get('telemetry.weeklyLastUseDate', StorageScope.GLOBAL, '0'));
-		const monthlyLastUseDate = Date.parse(storageService.get('telemetry.monthlyLastUseDate', StorageScope.GLOBAL, '0'));
+		const dailyLastUseDate: number = Date.parse(storageService.get('telemetry.dailyLastUseDate', StorageScope.GLOBAL, '0'));
+		const weeklyLastUseDate: number  = Date.parse(storageService.get('telemetry.weeklyLastUseDate', StorageScope.GLOBAL, '0'));
+		const monthlyLastUseDate: number  = Date.parse(storageService.get('telemetry.monthlyLastUseDate', StorageScope.GLOBAL, '0'));
+		const firstTimeUser: boolean = dailyLastUseDate === Date.parse('0');
 
-		let today = new Date().toUTCString();
+		let today: Date = new Date();
+		let todayString: string = new Date().toUTCString();
 
 		// daily user event
-		if (this.diffInDays(Date.parse(today), dailyLastUseDate) >= 1) {
+		if (this.diffInDays(Date.parse(todayString), dailyLastUseDate) >= 1) {
 			// daily first use
 			telemetryService.publicLog('telemetry.dailyFirstUse', { dailyFirstUse: true });
-			storageService.store('telemetry.dailyLastUseDate', today, StorageScope.GLOBAL);
+			storageService.store('telemetry.dailyLastUseDate', todayString, StorageScope.GLOBAL);
 		}
 
 		// weekly user event
-		if (this.diffInDays(Date.parse(today), weeklyLastUseDate) >= 7) {
+		if (this.diffInDays(Date.parse(todayString), weeklyLastUseDate) >= 7) {
 			// weekly first use
 			telemetryService.publicLog('telemetry.weeklyFirstUse', { weeklyFirstUse: true });
-			storageService.store('telemetry.weeklyLastUseDate', today, StorageScope.GLOBAL);
+			storageService.store('telemetry.weeklyLastUseDate', todayString, StorageScope.GLOBAL);
 		}
 
 		// monthly user events
-		if (this.diffInDays(Date.parse(today), monthlyLastUseDate) >= 30) {
+		if (this.diffInDays(Date.parse(todayString), monthlyLastUseDate) >= 30) {
 			telemetryService.publicLog('telemetry.monthlyUse', { monthlyFirstUse: true });
-			storageService.store('telemetry.monthlyLastUseDate', today, StorageScope.GLOBAL);
+			storageService.store('telemetry.monthlyLastUseDate', todayString, StorageScope.GLOBAL);
 		}
 
+		/* send monthly uses at the end of month */
+		const monthlyUseCount: number = storageService.getNumber('telemetry.monthlyUseCount', StorageScope.GLOBAL, 0);
+		let lastMonthDateString: string = storageService.get('telemetry.monthlyLastUseDate', StorageScope.GLOBAL, '0');
+		let lastMonthDate: Date = new Date(lastMonthDateString);
+		if (today.getFullYear() === lastMonthDate.getFullYear()) {
```

Comment:
This isn't going to roll over correctly to the new year - example if I used ADS for the first time on 12/31 I'd correctly have a count of 1 for monthlyUseCount stored. Then if I launched it on the 1st today.getFullYear would be the new year but lastMonthDay.getFullYear would still be last year - so this would skip this whole section until 30 days have passed since 12/31.